src/container.c

```diff
@@ -38,19 +40,44 @@ int container_start(void *arg) {
     return -1;
   }
 
+  int exit = 0;
+
+  char **argv = malloc((config->arg_length + 2 * sizeof(char *)));
+  if (argv == NULL) {
+    log_error("failed to allocate memory for argv: %m");
+    return -1;
+  }
+  int argv_print_length = 0;
+  argv[0] = config->cmd;
+  for (int i = 0; i < config->arg_length; i++) {
+    argv[i + 1] = config->arg[i];
+    argv_print_length += strlen(config->arg[i]) + 1;
+  }
+  // argv must be NULL terminated
+  argv[config->arg_length + 1] = NULL;
+  char *argv_echo_string = join_strings(argv, 1, config->arg_length);
+
   log_debug("executing command '%s %s' from directory '%s' in container...",
-            config->cmd, config->arg, config->mnt);
+            config->cmd, argv_echo_string, config->mnt);
   log_info("### BARCONTAINER STARTING - type 'exit' to quit ###");
-  // argv must be NULL terminated
-  char *argv[] = {config->cmd, config->arg, NULL};
+
   if (execve(config->cmd, argv, NULL)) {
     log_error("failed to execve '%s%s%s': %m", config->cmd,
               config->arg == NULL ? "" : " ",
-              config->arg == NULL ? "" : config->arg);
-    return -1;
+              config->arg == NULL ? "" : argv_print_length);
+    exit = -1;
+    goto cleanup;
   }
   log_debug("container started...");
 
+cleanup:
+  free(argv);
```

Comment:
We might want to do this in the function to stop/kill the container, that would make the container lifecycle easier to follow